plugin/type_uuid/item_uuidfunc.h

```diff
@@ -64,4 +64,35 @@ class Item_func_uuid: public Item_func_sys_guid
   { return get_item_copy<Item_func_uuid>(thd, this); }
 };
 
+class Item_func_uuid_v4: public Item_str_func
+{
```

Comment:
I suggest to derive from Item_func_sys_guid, like Item_func_uuid does.
It will help to avoid duplicating of a few methods:
- fix_length_and_dec
- const_item
- used_tables
- check_vcol_func_processor